### PR TITLE
blob/fileblob: ErrorAs supports os.PathError

### PR DESCRIPTION
**Pull request by jba**
Have ErrorAs recognize `os.PathError`s, since they are returned
from many filesystem operations.

### CODE REVIEW

File: blob/fileblob/fileblob.go [hunk 1]
```diff
@@ -403,7 +403,15 @@ func (b *bucket) ListPaged(ctx context.Context, opts *driver.ListOptions) (*driv
 func (b *bucket) As(i interface{}) bool { return false }
 
 // As implements driver.ErrorAs.
-func (b *bucket) ErrorAs(err error, i interface{}) bool { return false }
+func (b *bucket) ErrorAs(err error, i interface{}) bool {
```
Comment (vangent) [inline blob/fileblob/fileblob.go]: Please also update the package godoc for As.

File: blob/fileblob/fileblob_test.go [hunk 2]
```diff
@@ -168,3 +171,26 @@ func TestUnescape(t *testing.T) {
...
+func (verifyPathError) ListObjectCheck(o *blob.ListObject) error     { return nil }
+
+func (verifyPathError) ErrorCheck(b *blob.Bucket, err error) error {
+	var perr *os.PathError
+	if !b.ErrorAs(err, &perr) {
+		return errors.New("want ErrorAs to succeed for PathError")
+	}
+	const want = "/tmp/go-cloud-fileblob/key-does-not-exist"
+	if got := perr.Path; got != want {
+		return fmt.Errorf("got path %q, want %q", got, want)
```
Comment (vangent) [inline blob/fileblob/fileblob_test.go]: Looks like this failed on MacOS....

File: blob/fileblob/fileblob.go (hunk 1 above)
Comment (jba) [reply blob/fileblob/fileblob.go]: Done.

File: blob/fileblob/fileblob_test.go (hunk 2 above)
Comment (jba) [reply blob/fileblob/fileblob_test.go]: Hopefully fixed...

File: blob/fileblob/fileblob.go [hunk 3]
```diff
@@ -45,9 +45,10 @@
 //  - file:///c:/foo/bar
 //    -> Passes "c:/foo/bar".
 //
-// As
+// As and ErrorAs
 //
 // fileblob does not support any types for As.
+// It supports *os.PathError for ErrorAs
```
[the remark this answers is not in the document]
Comment (jba) [reply blob/fileblob/fileblob.go]: Done.

**Pull request merged**